Week6/app/src/main/java/com/em4n0101/mymovies/MovieDetailFragment.kt

```diff
@@ -21,17 +23,19 @@ class MovieDetailFragment : Fragment(R.layout.fragment_movie_detail) {
 
             // get view model
             moviesViewModel = ViewModelProvider(this).get(MoviesViewModel::class.java)
-            addObserver(args.movieTitleString)
+
+            getMovieByTitle(args.movieTitleString)
         }
     }
 
-    private fun addObserver(forMovieTitle: String) {
-        val observer = Observer<Movie> {
-            if (it != null) {
-                updateUIWith(it)
+    private fun getMovieByTitle(movieTitle: String) {
+        lifecycleScope.launch {
+            moviesViewModel.getMovieByTitleFlow(movieTitle).collect { movie ->
+                if (movie != null) {
+                    updateUIWith(movie)
+                }
```

Comment:
Same thing here. Collect the flow in the view model and expose data as LiveData value. I belive there are some extension functions on flow that you can use to convert it to live data. Not 100% sure though